### ISSUE TITLE
Testing components that use ApolloQuery 

### ISSUE THREAD

**Issue by dominik-bln** [Akryum/vue-apollo#609]
We were having some problems when testing components that make use of `ApolloQuery` in combination with `@vue/test-utils` because it's difficult to get the content of the scopedSlot to render.

With `shallowMount` and `ApolloQuery` on the top-level of a using component the render result is only `<apolloquery query="[object Object]" variables="[object Object]" debounce="200" skip="true" tag=""></apolloquery>`, so we opted to use `mount` like the following:

```javascript
const localVue = vueTestUtils.createLocalVue()
localVue.use(VueApollo)
const wrapper = mount(SomeComponentWithApolloQuery, {
  localVue
})
```

We then get `TypeError: Cannot read property 'loading' of undefined`, which is coming from the following line:

https://github.com/Akryum/vue-apollo/blob/2293adb03f23429af75723cc441ce01916699a83/src/components/ApolloQuery.js#L200

We can make it work by mocking `$apolloData`:

```javascript
const const mountConfig = mount({
   mocks: {
     $apolloData: {
       loading: false,
     },
  },
})
```

But it could be good to ensure `$apolloData.loading` is set in the plugin already if it relies on this data and to make sure others are not running into similar problems.

Thanks for the great work on this plugin!

**Comment by dominik-bln** [comment]
I'm happy to open an MR by the way, but am not completely sure if checking `$apolloData.loading` on the `ApollQuery` level is appropriate, or if the initialisation should be ensured on a lower level.

**Comment by P4sca1** [comment]
I'm getting the same error `[Vue warn]: Error in render: "TypeError: Cannot read property 'loading' of undefined"` in my `nuxt` project when trying to use `ApolloQuery`.
Is it working for you when not testing components?

**Comment by dominik-bln** [comment]
Yes, in the app it works fine. We are not using nuxt though, so not sure how to set it up there.

**Comment by P4sca1** [comment]
I noticed that the issue only occurs when rendering the component on the server.
Maybe this is related to your issue.

**Comment by P4sca1** [comment]
For me the issue was that I mounted apolloProvider wrong. Looking at your code it seems like you never mount an Apollo provider. Maybe there needs to be some kind of mock / testing Provider?

**Comment by tavo1987** [comment]
@P4sca1 do you have an example code about this?
